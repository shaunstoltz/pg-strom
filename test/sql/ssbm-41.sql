SET pg_strom.regression_test_mode = on;
SET search_path = pgstrom_regress,public;
SET pg_strom.debug_kernel_source = off;
--Q4_1
SET pg_strom.enabled = on;
select d_year, c_nation,  sum(lo_revenue - lo_supplycost) as profit
from date1, customer, supplier, part, lineorder
    where lo_custkey = c_custkey
       and lo_suppkey = s_suppkey
       and lo_partkey = p_partkey
       and lo_orderdate = d_datekey
       and c_region = 'AMERICA'
       and s_region = 'AMERICA'
       and (p_mfgr = 'MFGR#1' or p_mfgr = 'MFGR#2')
    group by d_year, c_nation
    order by d_year, c_nation;

SET pg_strom.enabled = off;
select d_year, c_nation,  sum(lo_revenue - lo_supplycost) as profit
from date1, customer, supplier, part, lineorder
    where lo_custkey = c_custkey
       and lo_suppkey = s_suppkey
       and lo_partkey = p_partkey
       and lo_orderdate = d_datekey
       and c_region = 'AMERICA'
       and s_region = 'AMERICA'
       and (p_mfgr = 'MFGR#1' or p_mfgr = 'MFGR#2')
    group by d_year, c_nation
    order by d_year, c_nation;
