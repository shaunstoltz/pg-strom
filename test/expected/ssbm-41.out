SET pg_strom.regression_test_mode = on;
SET search_path = pgstrom_regress,public;
SET pg_strom.debug_kernel_source = off;
--Q4_1
SET pg_strom.enabled = on;
select d_year, c_nation,  sum(lo_revenue - lo_supplycost) as profit
from date1, customer, supplier, part, lineorder
    where lo_custkey = c_custkey
       and lo_suppkey = s_suppkey
       and lo_partkey = p_partkey
       and lo_orderdate = d_datekey
       and c_region = 'AMERICA'
       and s_region = 'AMERICA'
       and (p_mfgr = 'MFGR#1' or p_mfgr = 'MFGR#2')
    group by d_year, c_nation
    order by d_year, c_nation;
 d_year |    c_nation     |   profit    
--------+-----------------+-------------
   1992 | ARGENTINA       | 21049817821
   1992 | BRAZIL          | 20793623985
   1992 | CANADA          | 21768888522
   1992 | PERU            | 21107265997
   1992 | UNITED STATES   | 22087350809
   1993 | ARGENTINA       | 21224064185
   1993 | BRAZIL          | 20214018274
   1993 | CANADA          | 21487793123
   1993 | PERU            | 20451314966
   1993 | UNITED STATES   | 21537297778
   1994 | ARGENTINA       | 21492078723
   1994 | BRAZIL          | 20119322546
   1994 | CANADA          | 21245670138
   1994 | PERU            | 20925002153
   1994 | UNITED STATES   | 21768994510
   1995 | ARGENTINA       | 21744240441
   1995 | BRAZIL          | 20626871263
   1995 | CANADA          | 21970480412
   1995 | PERU            | 21358554139
   1995 | UNITED STATES   | 21533231465
   1996 | ARGENTINA       | 21906426061
   1996 | BRAZIL          | 21091679462
   1996 | CANADA          | 21089650746
   1996 | PERU            | 20849385079
   1996 | UNITED STATES   | 21245236259
   1997 | ARGENTINA       | 21906122939
   1997 | BRAZIL          | 20930283186
   1997 | CANADA          | 21478124066
   1997 | PERU            | 21254050870
   1997 | UNITED STATES   | 21818581630
   1998 | ARGENTINA       | 12360150387
   1998 | BRAZIL          | 11535854687
   1998 | CANADA          | 12711817151
   1998 | PERU            | 12175877989
   1998 | UNITED STATES   | 12731502856
(35 rows)

SET pg_strom.enabled = off;
select d_year, c_nation,  sum(lo_revenue - lo_supplycost) as profit
from date1, customer, supplier, part, lineorder
    where lo_custkey = c_custkey
       and lo_suppkey = s_suppkey
       and lo_partkey = p_partkey
       and lo_orderdate = d_datekey
       and c_region = 'AMERICA'
       and s_region = 'AMERICA'
       and (p_mfgr = 'MFGR#1' or p_mfgr = 'MFGR#2')
    group by d_year, c_nation
    order by d_year, c_nation;
 d_year |    c_nation     |   profit    
--------+-----------------+-------------
   1992 | ARGENTINA       | 21049817821
   1992 | BRAZIL          | 20793623985
   1992 | CANADA          | 21768888522
   1992 | PERU            | 21107265997
   1992 | UNITED STATES   | 22087350809
   1993 | ARGENTINA       | 21224064185
   1993 | BRAZIL          | 20214018274
   1993 | CANADA          | 21487793123
   1993 | PERU            | 20451314966
   1993 | UNITED STATES   | 21537297778
   1994 | ARGENTINA       | 21492078723
   1994 | BRAZIL          | 20119322546
   1994 | CANADA          | 21245670138
   1994 | PERU            | 20925002153
   1994 | UNITED STATES   | 21768994510
   1995 | ARGENTINA       | 21744240441
   1995 | BRAZIL          | 20626871263
   1995 | CANADA          | 21970480412
   1995 | PERU            | 21358554139
   1995 | UNITED STATES   | 21533231465
   1996 | ARGENTINA       | 21906426061
   1996 | BRAZIL          | 21091679462
   1996 | CANADA          | 21089650746
   1996 | PERU            | 20849385079
   1996 | UNITED STATES   | 21245236259
   1997 | ARGENTINA       | 21906122939
   1997 | BRAZIL          | 20930283186
   1997 | CANADA          | 21478124066
   1997 | PERU            | 21254050870
   1997 | UNITED STATES   | 21818581630
   1998 | ARGENTINA       | 12360150387
   1998 | BRAZIL          | 11535854687
   1998 | CANADA          | 12711817151
   1998 | PERU            | 12175877989
   1998 | UNITED STATES   | 12731502856
(35 rows)

